### PR TITLE
Add sticky scroll into interactive editor code block that respects the editor sticky scroll setting

### LINKED ISSUE #227644: Enable sticky scroll in chat code blocks
From https://github.com/microsoft/vscode-dotnettools/issues/1361

### PR DESCRIPTION
**Pull request by aiday-mar**
Add sticky scroll into interactive editor code block that respect the editor sticky scroll setting

fixes https://github.com/microsoft/vscode/issues/227644

https://github.com/user-attachments/assets/40358546-6f7d-4389-aef5-edb65e7deefb

### CODE REVIEW

**Comment by mjbvz** [comment]
Thanks for taking a look! How were you able to shrink the code block's height so that sticky scroll shows up? I always see the code block rendering without any vertical scroll bars

I was also imagining that the sticky scrolling would be based on the scrolling of the chat itself, not the editor. Is that the case?

File: src/vs/workbench/contrib/chat/browser/chatOptions.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import { Color } from '../../../../base/common/color.js';
 import { Emitter } from '../../../../base/common/event.js';
 import { Disposable } from '../../../../base/common/lifecycle.js';
-import { IBracketPairColorizationOptions, IEditorOptions } from '../../../../editor/common/config/editorOptions.js';
+import { IBracketPairColorizationOptions, IEditorOptions, IEditorStickyScrollOptions } from '../../../../editor/common/config/editorOptions.js';
```
Comment (Petr2901) [inline src/vs/workbench/contrib/chat/browser/chatOptions.ts:9]: ahoj

**Comment by aiday-mar** [comment]
Hi Matt, thanks a lot for looking into this. Those are very good questions and I forgot to discuss them with you. As you mentioned I have also seen the code block always rendered with its full height. To make the example work, I force a dimension of 100 by 100 pixels in the constructor which made the vertical scroll bar appear when generating a lot of code. I guess I assumed that what you wanted to do is to at some point cap the editor height to a specific maximum (like say 1000 pixels) and past that point if one generates long code, there would be a vertical scroll bar and you could see the editor sticky scroll. It is true that as of now this PR adds no functionality because you never see it action. 

> I was also imagining that the sticky scrolling would be based on the scrolling of the chat itself, not the editor. Is that the case?

The way the editor sticky scroll works is that it is based on the scrolling of the editor itself, not the scroll of the surrounding container. We could work on making the sticky scroll depend on the scroll position of the surrounding container, that would require some more work and I wonder if that is what we want?

**Comment by aiday-mar** [comment]
Closing this PR as likely will not be merged

**Pull request closed without merging**